addons/payment_mercado_pago/models/payment_transaction.py

```diff
@@ -0,0 +1,272 @@
+                create_refund_transaction=create_refund_transaction,
+            )
+
+        refund_tx = super()._send_refund_request(
+            amount_to_refund=amount_to_refund, create_refund_transaction=True
+        )
+
+        # Get the internal reference of the payment to refund.
+        payment_id = self.acquirer_reference
+        if not payment_id:
+            raise ValidationError(
+                "Mercado Pago" + _(
+                    "could not retrieve a payment id for the transaction with reference %s",
+                    self.reference
+                )
+            )
+
+        data = self.acquirer_id._mercado_pago_make_request(
+            f'/v1/payments/{payment_id}/refunds', {'amount': amount_to_refund}, 'POST'
+        )
+
+        # Note that for MP a refund is not a transaction, but a refund object linked to a payment.
+        # Therefore for a refund tx the acquirer reference is the refund id.
+        # To retrieve the payment we search for the source tx and its acquirer reference is the
+        # the reference to the source payment.
+        refund_acquirer_reference = data.get('id')
+        if not refund_acquirer_reference:
+            raise ValidationError("Mercado Pago:" + _("no refund id returned, bad response."))
+        refund_tx.acquirer_reference = refund_acquirer_reference
+
+        _logger.info(
+            "Mercado Pago: refund request response for transaction wih reference %s:\n%s",
+            self.reference, pprint.pformat(data)
+        )
+
+        refund_tx._handle_notification_data('mercado_pago', data)
+
+        return refund_tx
+
+    def _get_tx_from_notification_data(self, provider, notification_data):
+        """ Override of payment to find the transaction based on Mercado Pago data.
+
+        :param str provider: The provider of the acquirer that handled the transaction
+        :param dict notification_data: The notification data sent by the provider
+        :return: The transaction if found
+        :rtype: recordset of `payment.transaction`
+        :raise: ValidationError if inconsistent data were received
+        :raise: ValidationError if the data match no transaction
+        """
+        tx = super()._get_tx_from_notification_data(provider, notification_data)
+
+        if provider != 'mercado_pago' or len(tx) == 1:
+            return tx
+
+        reference = notification_data.get('external_reference')
+        tx_id = notification_data.get('tx_id')
+
+        if not reference and not tx_id:
+            raise ValidationError(
+                "Mercado Pago: " +
+                _(
+                    "received data with missing merchant external reference and missing "
+                    "transaction id."
+                )
+            )
+
+        if reference:
+            tx = self.search([('reference', '=', reference), ('provider', '=', 'mercado_pago')])
+        else:
+            tx = self.search([('id', '=', tx_id)])
```

Comment:
```suggestion
            tx = self.search([('id', '=', tx_id), ('provider', '=', mercado_pago)])
```
to be sure early in the flow that somebody is not trying stuff on transactions of other acquirers.